components/director/internal/domain/application/service.go

```diff
@@ -488,6 +497,35 @@ func (s *service) DeleteLabel(ctx context.Context, applicationID string, key str
 	return nil
 }
 
+// IsBeingUsed Checks if an application has scenarios associated with it. If the scenarios are part of a runtime, then the application is considered being used by that runtime.
+func (s *service) IsBeingUsed(ctx context.Context, tenant, id string) (bool, error) {
```

Comment:
thats a weird name - maybe something like IsApplicationInScenarios? and then id arg to be called appID ?